src/server/controllers/buzz.controller.ts

```diff
@@ -119,11 +126,75 @@ export function createBuzzTipTransactionHandler({
     if (fromAccountId === input.toAccountId)
       throw throwBadRequestError('You cannot send buzz to the same account');
 
-    return createBuzzTransaction({
+    const { entityType, entityId } = input;
+    let targetUserIds: number[] = [input.toAccountId].filter(isDefined);
+
+    if ((entityType === 'Post' || entityType === 'Image') && entityId) {
+      // May have contributros, check this...
+      const collaboratorEntityType = EntityType.Post; // For the time being, only this is supported.
+      const collaboratorEntityId =
+        entityType === 'Post' ? entityId : (await getImageById({ id: entityId }))?.postId;
+
+      if (collaboratorEntityId && collaboratorEntityType) {
+        const collaborators = await getEntityCollaborators({
+          entityId: collaboratorEntityId,
+          entityType: collaboratorEntityType,
+        });
+
+        const collaboratorIds = collaborators.map((c) => c.user.id);
+
+        targetUserIds = [...new Set([...targetUserIds, ...collaboratorIds])].filter(isDefined);
+      }
+    }
+
+    const amount = Math.floor(input.amount / targetUserIds.length);
```

Comment:
just to be extra safe here, don't think it'll ever happen
```suggestion
    const amount = Math.floor(input.amount / (targetUserIds.length || 1));
```

Reply:
Added a throw error above if no target ids are available.